### PR TITLE
blaze: just make the kernel depend on //sys/src/cmd:cmds

### PR DESCRIPTION
**Pull request by rminnich**
blaze is so much smarter than our old build we can just do it this way.
One issue: this fails with a very unclear diagnostic, I have to run it twice.
Help?

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by sevki** [comment]
since the goal of this thing is not to generate anything in the tree, this will not work without this slight modification. 

As for the reason why this modification is necessary, dependencies install stuff to their dependants, `cc_binary` installs it's binary to `bin/foo` in everything that has defined foo as a dependency. In Unix systems this install is a symlink, in harvey this is a bind. So the `config` rule, looking at its deps knows that under `bin/` there will be `bin/echo` `bin/bind` and so on. 

This is great because we symlink whatever is being installed(binary, library, mksys generated file) to the exact hash of the dependency that we need.
For instance, when building libc
```
├── libc-623dc7b991af638b7474f6f934fbb3475bf6476d
│   ├── 9syscall
│   │   ├── alarm.s -> /tmp/build/9syscall-a3d245729eebc95682d7160018971db36a07a430/alarm.s
│   │   ├── await.s -> /tmp/build/9syscall-a3d245729eebc95682d7160018971db36a07a430/await.s
│   │   ├── bind.s -> /tmp/build/9syscall-a3d245729eebc95682d7160018971db36a07a430/bind.s
```
we can see that all these items are bound to a specific hash. As a result, we don't need to do a `make clean` type operation before every build because we know we are always using right thing version. 

Anyways, here is the "slight" modification, 

sys/src/cmd/BUILD:
```python
CMDS = [
        "//sys/src/cmd/acme:acme",
        "//sys/src/cmd/rc:rc",
        "//sys/src/cmd/ip/ipconfig:ipconfig",
        ":echo",
        ":mount",
        ":date",
        ":srv",
        ":ls",
        ":bind",
]
group(
    name="cmds",
    deps=CMDS,
)
```
sys/src/9/amd64/BUILD:
```python
load("//sys/src/cmd/BUILD", "CMDS")
config(
    name="amd64cpu",
    deps=CMDS,
    ...
)

```

**Comment by rminnich** [comment]
do you want to change this CL with that modification or should I? I'm still a bit confused by all this so maybe it should be you.

**Comment by sevki** [comment]
sure no problem.

File: sys/src/9/amd64/BUILD [hunk 1]
```diff
@@ -377,12 +378,7 @@ cc_binary(
         ":systab",
         ":sys",
         ":inith",
-        "//sys/src/libmemlayer:libkmemlayer",
-        "//sys/src/libmemdraw:libkmemdraw",
-        "//sys/src/libdraw:libkdraw",
-        "//sys/src/libc:libkc",
-        "//sys/src/libip:libkip",
-        "//sys/src/libsec:libksec",
+        "//sys/src:klibs",
```
Comment (sevki) [inline sys/src/9/amd64/BUILD]: groups issue

File: sys/src/9/amd64/BUILD [hunk 2]
```diff
@@ -525,7 +493,7 @@ cc_binary(
         "-mcmodel=small",
     ],
     deps=[
-        "//sys/src/libc:libc"
+        "//sys/src:libs",
```
Comment (sevki) [inline sys/src/9/amd64/BUILD]: groups issue

**Comment by sevki** [comment]
you are getting this issue because it thinks that libs is a library called s, you need to have the libraries listed there individually, or you can do the same thing you did with cmds and have it as an array in sys/src/BUILD

**Comment by sevki** [comment]
LGTM

**Pull request merged**